les/fetcher.go

```diff
@@ -219,8 +216,12 @@ func (f *lightFetcher) syncLoop() {
 			p.Log().Debug("Done synchronising with peer")
 			f.checkSyncedHeaders(p)
 			f.syncing = false
+			r := f.requesting
```

Comment:
If client is syncing, then the `f.requesting` MUST be false since fetcher will ignore all following fetching request.
Let's simplify it by:

```go
f.lock.Lock()
p.Log().Debug("Done synchronising with peer")
f.checkSyncedHeaders(p)
f.syncing = false	
f.requesting = true		
f.lock.Unlock()
f.requestTrigger <- struct{}{}
```